lms/static/scripts/frontend_apps/components/dashboard/CourseActivity.tsx

```diff
@@ -19,23 +27,37 @@ type AssignmentsTableRow = {
   replies: number;
 };
 
-const assignmentURL = (id: number) => urlPath`/assignments/${String(id)}`;
-
 /**
  * Activity in a list of assignments that are part of a specific course
  */
 export default function CourseActivity() {
   const { courseId } = useParams<{ courseId: string }>();
+  const [, navigate] = useLocation();
   const { dashboard } = useConfig(['dashboard']);
   const { routes } = dashboard;
+
+  const { filters, updateFilters } = useDashboardFilters();
+  const { assignmentIds, studentIds } = filters;
+  const search = useSearch();
+  const hasSelection = assignmentIds.length > 0 || studentIds.length > 0;
+  const onClearSelection = useCallback(
+    // Clear every filter but courses
+    () => updateFilters({ studentIds: [], assignmentIds: [] }),
+    [updateFilters],
+  );
+
   const course = useAPIFetch<Course>(
     replaceURLParams(routes.course, { course_id: courseId }),
   );
   const assignments = useAPIFetch<AssignmentsResponse>(
     replaceURLParams(routes.course_assignments_metrics, {
       course_id: courseId,
     }),
-    { public_id: dashboard.organization_public_id },
+    {
+      assignment_id: assignmentIds,
+      h_userid: studentIds,
+      public_id: dashboard.organization_public_id,
```

Comment:
Unrelated to this PR, I suggest we give `public_id` a better name. At present it isn't obvious that it refers to an _organization_ ID as opposed to a public for an assignment.